tachyon/math/matrix/matrix_utils.h

```diff
@@ -45,22 +45,22 @@ MakeCirculant(const Eigen::MatrixBase<ArgType>& arg) {
                                  CirculantFunctor<ArgType>(arg.derived()));
 }
 
-template <typename PackedPrimeField, typename Derived, typename PrimeField>
-std::vector<PackedPrimeField> PackRowHorizontally(
-    const Eigen::MatrixBase<Derived>& matrix, size_t row,
-    std::vector<PrimeField>& remaining_values) {
+template <typename PackedPrimeField, typename PrimeField>
+std::vector<PackedPrimeField*> PackRowHorizontally(
+    Eigen::Block<RowMajorMatrix<PrimeField>>& matrix, size_t row,
```

Comment:
Can't `matrix` be const reference?

Reply:
Both `remaining_values` and the return value `reinterpret_cast`s the elements of matrix, meaning that if `remaining_values` or the return value is changed, the matrix changes itself too. So it should not be `const`!